configobject/history/history.go

```diff
@@ -34,23 +34,19 @@ var mysqlObservers = struct {
 }
 
 var historyCounter = struct {
-	state            int
-	notification     int
-	usernotification int
-	downtime         int
-	comment          int
-	flapping         int
-	acknowledgement  int
+	state            int64
+	notification     int64
+	usernotification int64
+	downtime         int64
+	comment          int64
+	flapping         int64
+	acknowledgement  int64
 }{}
 
-var historyCounterLock = sync.Mutex{}
-
-func printAndResetHistoryCounter(counter *int, historyType string) {
+func printAndResetHistoryCounter(counter *int64, historyType string) {
 	if *counter > 0 {
 		log.Infof("Added %d %s history entries in the last 20 seconds", *counter, historyType)
-		historyCounterLock.Lock()
-		*counter = 0
-		historyCounterLock.Unlock()
+		atomic.SwapInt64(counter, 0)
```

Comment:
Skywalker would say: Every line in this change is wrong.

Reply:
should i undo it then